src/core/public/apm_system.ts

```diff
@@ -87,6 +74,45 @@ export class ApmSystem {
     });
   }
 
+  /* Hold the page load transaction open, until all resources actually finish loading */
+  private holdPageLoadTransaction(apm: ApmBase) {
+    this.pageLoadTransaction = apm.getCurrentTransaction();
+
+    // Keep the page load transaction open until all resources finished loading
+    if (this.pageLoadTransaction && this.pageLoadTransaction.type === 'page-load') {
+      // @ts-expect-error 2339
+      this.pageLoadTransaction.block(true);
+      this.pageLoadTransaction.mark('apmSetup');
+    }
+  }
+
+  /* Returns the number of resources that were loaded from network (vs. from disk cache) */
+  private getLoadedResourcesCount() {
+    try {
+      return window.performance
```

Comment:
This would be slow as it returns all the Resource timing info at a given time and includes a large amount of resources. It would be better to use the https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver API and use that to capture the information.